File: 1829_maximum_xor_for_each_query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int mask=(1<<maximumBit)-1;
        for(int i=0, x=0; i<nums.size(); i++) x^=nums[i], nums[i]=x^mask;
        reverse(nums.begin(), nums.end());
        return nums;
    }
};

File: 1829_maximum_xor_for_each_query.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int mask = (1 << maximumBit) - 1, acc = 0, len = nums.length;
        int[] res = new int[len];
        for (int n : nums) acc ^= n;
        for (int i = 0; i < len; i++) {
            res[i] = acc ^ mask;
            acc ^= nums[len - 1 - i];
        }
        return res;
    }
}
